### DISCUSSION TITLE
Welcome to giu Discussions!

### DISCUSSION THREAD

**Discussion by AllenDang** [AllenDang/giu#90] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by neutralinsomniac** [comment]
Hi! I came across your framework today and am looking to port my existing exocortex program here: https://github.com/neutralinsomniac/exocortex to giu! So far I love how simple and fast it is to use, but I've hit my first roadblock. I'm trying to build a dynamic list of buttons, where each button has different data attached to it, but I'm not seeing the correct way to do it. Here is the code I have right now: https://github.com/neutralinsomniac/exocortex/blob/3dc33e8add084b75d6d0d63114be6627055edae1/cmd/exogiu/exogiu.go#L44

With this code, every button gets the last tag in the list passed to it, instead of a different tag per button. Do you know of a smarter way to handle this situation?

Thank you!

**Reply by neutralinsomniac** [reply]
Nevermind! A friend of mine pointed me to where I was wrong: I was writing the closure over the tag iterator variable instead of making a non-iterating copy. Anyway, thank you for this amazing library!

**Reply by AllenDang** [reply]
Thanks for using my lib, :P

**Reply by AllenDang** [reply]
That is a common issue when creating in a loop. I've created a RangeBuilder to batch create widgets. I think I should add it to wiki.
